frontend/src/App.tsx

```diff
@@ -12,6 +13,7 @@ export default function App() {
                 <Route path={"/login"} element={<WelcomePage/>}/>
                 <Route path={"/register"} element={<RegisterPage/>}/>
                 <Route path={"/"} element={<MainPage/>}/>
+                <Route path={"/:id"} element={<InfoPage/>}/>
```

Comment:
here you'll get into trouble later on. What will happen if the id is `register`? 
```suggestion
                <Route path={"/login"} element={<WelcomePage/>}/>
                <Route path={"/register"} element={<RegisterPage/>}/>
                <Route path={"/"} element={<MainPage/>}/>
                <Route path={"/infos/:id"} element={<InfoPage/>}/>
```